x-pack/plugin/snapshot-repo-test-kit/qa/azure/src/javaRestTest/java/org/elasticsearch/repositories/blobstore/testkit/AzureSnapshotRepoTestKitIT.java

```diff
@@ -27,36 +29,40 @@ public class AzureSnapshotRepoTestKitIT extends AbstractSnapshotRepoTestKitRestT
     private static final String AZURE_TEST_CONTAINER = System.getProperty("test.azure.container");
     private static final String AZURE_TEST_KEY = System.getProperty("test.azure.key");
     private static final String AZURE_TEST_SASTOKEN = System.getProperty("test.azure.sas_token");
-
-    private static AzureHttpFixture fixture = new AzureHttpFixture(
+    private static final String AZURE_TEST_TENANT_ID = System.getProperty("test.azure.tenant_id");
+    private static final String AZURE_TEST_CLIENT_ID = System.getProperty("test.azure.client_id");
+    private static final AzureHttpFixture fixture = new AzureHttpFixture(
         USE_FIXTURE
             ? ESTestCase.inFipsJvm() ? AzureHttpFixture.Protocol.HTTP : AzureHttpFixture.Protocol.HTTPS
             : AzureHttpFixture.Protocol.NONE,
         AZURE_TEST_ACCOUNT,
         AZURE_TEST_CONTAINER,
-        Strings.hasText(AZURE_TEST_KEY) || Strings.hasText(AZURE_TEST_SASTOKEN)
-            ? AzureHttpFixture.sharedKeyForAccountPredicate(AZURE_TEST_ACCOUNT)
-            : AzureHttpFixture.MANAGED_IDENTITY_BEARER_TOKEN_PREDICATE
+        AZURE_TEST_TENANT_ID,
+        AZURE_TEST_CLIENT_ID,
+        decideAuthHeaderPredicate()
     );
 
-    private static TestTrustStore trustStore = new TestTrustStore(
+    private static Predicate<String> decideAuthHeaderPredicate() {
+        if (Strings.hasText(AZURE_TEST_KEY) || Strings.hasText(AZURE_TEST_SASTOKEN)) {
+            return AzureHttpFixture.sharedKeyForAccountPredicate(AZURE_TEST_ACCOUNT);
+        } else if (Strings.hasText(AZURE_TEST_TENANT_ID) && Strings.hasText(AZURE_TEST_CLIENT_ID)) {
+            return AzureHttpFixture.WORK_IDENTITY_BEARER_TOKEN_PREDICATE;
+        } else {
```

Comment:
else let's assert that neither `AZURE_TEST_TENANT_ID` nor `AZURE_TEST_CLIENT_ID` are set.